fahrschein-opentelemetry/src/main/java/org/zalando/fahrschein/opentelemetry/InstrumentedNakadiPublisher.java

```diff
@@ -0,0 +1,46 @@
+package org.zalando.fahrschein.opentelemetry;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import org.zalando.fahrschein.NakadiPublisher;
+
+import java.io.IOException;
+import java.util.List;
+
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER;
+
+public final class InstrumentedNakadiPublisher {
+
+    private final NakadiPublisher client;
+    private final Tracer tracer;
+
+    public InstrumentedNakadiPublisher(NakadiPublisher client, Tracer tracer) {
+        this.client = client;
+        this.tracer = tracer;
+    }
+
+    public <T> void publish(String eventName, List<T> events, Span parentSpan) throws IOException {
```

Comment:
Up to debate: apparently opentelemetry prefers inferring the parent span from the "activeSpan" context. From my understanding this only works in a thread-scoped execution flow, so I would rather prefer explicitly passing the parent here.

Reply:
@MALPI I've been reading a bit more about OpenTelemetry. They are really trying hard to be hidden from the actual business APIs, I think it makes sense to follow this convention. If people are running in a task-based instead of thread-based environment, they'll need to create a parent span around this call in order to preserve the context (or find a different way to set the active parent span).